src/app/utils/field-validation/validators/decimalValidator.ts

```diff
@@ -0,0 +1,63 @@
+import { chain, left, right } from 'fp-ts/lib/Either'
+import { flow } from 'fp-ts/lib/function'
+import isFloat from 'validator/lib/isFloat'
+import isInt from 'validator/lib/isInt'
+
+import { IDecimalField } from 'src/types/field'
+import { ResponseValidator } from 'src/types/field/utils/validation'
+import { ISingleAnswerResponse } from 'src/types/response'
+
+import { notEmptySingleAnswerResponse } from './common'
+
+type DecimalValidator = ResponseValidator<ISingleAnswerResponse>
+type DecimalValidatorConstructor = (
+  decimalField: IDecimalField,
+) => DecimalValidator
+
+const makeDecimalFloatRangeValidator: DecimalValidatorConstructor = (
+  decimalField,
+) => (response) => {
+  const { customMin, customMax } = decimalField.ValidationOptions || {}
+  const { answer } = response
+  interface IIsFloatOptions {
+    min?: number
+    max?: number
+  }
```

Comment:
don't declare interfaces in a running function

Reply:
shifted out